File: src/main/org/nlogo/prim/etc/_outputwrite.java
package org.nlogo.prim.etc ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Syntax;
import org.nlogo.nvm.Workspace;

public final strictfp class _outputwrite
	extends org.nlogo.nvm.Command
{
	@Override
	public void perform( final org.nlogo.nvm.Context context ) throws LogoException
	{
		workspace.outputObject
			( args[ 0 ].report( context ) ,
			  null , false , true ,
			  Workspace.OutputDestination.OUTPUT_AREA ) ;
		context.ip = next ;
	}
	@Override
	public Syntax syntax()
	{
		int[] right = { Syntax.TYPE_READABLE } ;
		return Syntax.commandSyntax( right );
	}
}

File: src/main/org/nlogo/prim/etc/_outputwrite.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.{ Syntax, OutputDestination }
import org.nlogo.nvm.{ Command, Context }

class _outputwrite extends Command {
  override def syntax =
    Syntax.commandSyntax(Array(Syntax.ReadableType))
  override def perform(context: Context) {
    workspace.outputObject(
      args(0).report(context), null, false, true,
      OutputDestination.OutputArea)
    context.ip = next
  }
}
